src/log.c

```diff
@@ -56,6 +56,40 @@ static int fd_set_block(int fd) {
 }
 #endif /* PR_USE_NONBLOCKING_LOG_OPEN */
 
+void pr_log_stacktrace(const char*moduleversion, const char*function) {
+#if defined(HAVE_EXECINFO_H) && \
+    defined(HAVE_BACKTRACE) && \
+    defined(HAVE_BACKTRACE_SYMBOLS)
+  void *trace[PR_TUNABLE_CALLER_DEPTH];
+  char **strings;
+  int tracesz;
+
+  (void) pr_log_debug(DEBUG4, "%s: -----BEGIN STACK TRACE in %s -----",moduleversion, function);
```

Comment:
I would probably change this implementation slightly, such that the debug level (if used/provided) could be configured by the caller.  Although one might argue that stacktraces are _always_ important, and thus perhaps `pr_log_pri(PR_LOG_NOTICE, ...)` might be a better fit.